R/caption.R

```diff
@@ -0,0 +1,110 @@
+#' Goal is to preserve space in between short and main caption
+#' when short is repeated. If short not repeated, then ok to just trim.
+#' trimws() is called quite a bit, but important to consider context of the
+#' call.
+#'
+#' @param text a chunk of caption text that may or may not need to be parsed.
+#' @param short optional short caption text
+#' @noRd
+parse_caption <- function(text, short = NULL, short_repeat = TRUE, short_sep = "") {
+  text <- trimws(text)
+  parsed_short <- str_extract(text, "^\\[.+?\\]")
```

Comment:
So the two limitations of the "[short] ..." syntax I can think of are

 * short text can't contain a new line character
 * short text can't contain a `]`

For the first, there may not be any cases where it would reasonable for short text to contain a new line (and a single plain `\n` won't be rendered differently by LaTeX anyway).

For the second, I guess there could be some cases, but then you can use the `short` argument.

Reply:
Pushed some updates so that parser can handle newline and sensible embedded brackets.

Reply:
Nice, I just meant to note the limitations given that the `short` arg could be used, but your proposed approach looks good to me.